src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,46 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    Animal.alive.push(this);
+  }
+
+  die() {
+    const index = Animal.alive.indexOf(this);
+
+    if (index > -1) {
+      Animal.alive.splice(index, 1);
```

Comment:
Instead of using `splice` to remove the animal from the `Animal.alive` array, it is preferred to use the `filter` method to remove animals with health less than or equal to 0. This approach is more aligned with the checklist guidelines.